pkg/cmd/root/root.go

```diff
@@ -55,6 +62,15 @@ func NewRootCommand(f *factory.Factory, version string) *cobra.Command {
 	cmd.AddCommand(cliversion.NewVersionCmd(f))
 	cmd.AddCommand(config.NewConfigCommand(f))
 
+	loc, err := pluginloc.New(nil)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+
+	pfact := pluginfactory.New(build.Version, loc)
```

Comment:
So idea of that factory is to be able to mock and inject different implementations to CLi in order to test it. 
Here we reinstantiate factory with version and localization. My take is that we need additional wrapper function that takes `loc` file from the plugin rather than require us to pass it. Apart from that this is pretty much what needs to be done. 

This would work as probably both host and plugin use the same configs (@namit-chandwani would need to refactor it to represents real state of the matters)